contracts/peripheral/grants/VeBalGrant.sol

```diff
@@ -0,0 +1,352 @@
+     * @param _value   Eth to be sent
+     * @param _data    Call data
+     */
+    function execute(
+        address _to,
+        uint256 _value,
+        bytes memory _data
+    ) external returns (bool, bytes memory) {
+        active ? require(msg.sender == project, "!caller") : require(msg.sender == balancer, "!caller");
+        require(
+            _to != address(WETH) && _to != address(BAL) && _to != address(BAL_ETH_BPT) && _to != address(votingEscrow),
+            "invalid target"
+        );
+
+        bytes4 sig;
+        assembly {
+            sig := mload(add(_data, 32))
+        }
+
+        require(sig != IFeeDistributor.claimToken.selector, "!allowed");
+
+        (bool success, bytes memory result) = _to.call{ value: _value }(_data);
+        require(success, "!success");
+        return (success, result);
+    }
+
+    /* ----------------------------------------------------------------
+       Project Functions
+    ---------------------------------------------------------------- */
+
+    /**
+     * @notice Increase amount locked in veBAL using BPT balance of contract
+     * @notice Only the project may call this while the grant is active
+     * @param amount number of BPT tokens to lock
+     */
+    function increaseLock(uint256 amount) public onlyProject whileActive {
+        _increaseLock(amount);
+    }
+
+    /**
+     * @notice Increase veBAL lock time
+     * @notice Only the project may call this while the grant is active
+     * @param to the new unlock time
+     */
+    function increaseTime(uint256 to) external onlyProject whileActive {
+        votingEscrow.increase_unlock_time(to);
+    }
+
+    /**
+     * @notice claim fees from distributor
+     * @notice Locks as BPT if the fee is bal or weth
+     * @notice Sends if token is not weth or bal
+     * @notice Only the project may call this while the grant is active
+     * @param distro fee distributor being called
+     * @param token token being claimed from distributor
+     * @param to receiver in the send situation
+     */
+    function claimFees(
+        address distro,
+        address token,
+        address to
+    ) external onlyProject whileActive {
+        IFeeDistributor(distro).claimToken(address(this), IERC20(token));
+
+        if (token == address(BAL) || token == address(WETH)) {
+            _joinBalEthPool();
+            uint256 _balance = BAL_ETH_BPT.balanceOf(address(this));
+            _increaseLock(_balance);
+        } else {
+            require(to != address(0), "!0");
+            IERC20(token).transfer(to, IERC20(token).balanceOf(address(this)));
+        }
+    }
+
+    /**
+     * @notice creates the initial lock for the grant
+     * @notice tracks lock state and weth contributed by project
+     * @notice Only the project may call this while the grant is active
+     * @param unlockTime When the lock will be lifted
+     */
+    function createLock(uint256 unlockTime) external onlyProject whileActive {
```

Comment:
Should be there any validation on `unlockTime` ? Future time, or minimum period ? 
Should be there a validation that balance is not zero ? 
Or all that is handled somehow on the `        votingEscrow.create_lock(balance, unlockTime);
` ?

Reply:
Lets keep it simple, we don't need to worry about initial unlock time.